### PR TITLE
[BugFix][310p] Align 310P GDN state semantics with vLLM

### PR DESCRIPTION
**Pull request by Tflowers-0129**
### What this PR does / why we need it?

- I found that the FLA state semantics are not aligned between vllm-ascend and vLLM: in vLLM, the state layout is `[N, HV, V, K]`, while in vllm-ascend it is `[N, HV, K, V]`. The early 310P implementation was aligned with the mainline vllm-ascend behavior, but the operator team now plans to provide a ` fused recurrent-gated-delta-rule` operator whose semantics will align with vLLM. Therefore, this needs to be adjusted in advance.

- I also found that L2 normalization can use the operator provided in `functional`, instead of being fully implemented through manually stitched small ops as it is now.

### Does this PR introduce _any_ user-facing change?
NA

### How was this patch tested?
ut and the e2e test

- vLLM version: v0.18.0
- vLLM main: https://github.com/vllm-project/vllm/commit/35141a7eeda941a60ad5a4956670c60fd5a77029

### CODE REVIEW

File: vllm_ascend/_310p/ops/fla/fused_recurrent_gated_delta_rule.py [hunk 1]
```diff
@@ -1,10 +1,11 @@
 import torch
+import torch.nn.functional as F
```
[the remark this answers is not in the document]
Comment (Tflowers-0129) [reply vllm_ascend/_310p/ops/fla/fused_recurrent_gated_delta_rule.py:2]: updated

File: vllm_ascend/_310p/ops/fla/chunk_gated_delta_rule.py [hunk 2]
```diff
@@ -104,12 +100,12 @@ def _torch_chunk_gated_delta_rule_chunked(
...
     g/beta:    [B, T, H]
-    initial_state: [B, H, K, V]
+    initial_state: [B, H, V, K]
     """
     initial_dtype = query.dtype
     if use_qk_l2norm_in_kernel:
-        query = _l2norm(query, dim=-1, eps=1e-6)
-        key = _l2norm(key, dim=-1, eps=1e-6)
+        query = F.normalize(query, p=2, dim=-1).to(query.dtype)
+        key = F.normalize(key, p=2, dim=-1).to(key.dtype)
```
[the remark this answers is not in the document]
Comment (Tflowers-0129) [reply vllm_ascend/_310p/ops/fla/chunk_gated_delta_rule.py]: updated, i have used the :  F.normalize(key, p=2, dim=-1, eps=1e-6).to(key.dtype)

File: vllm_ascend/_310p/ops/fla/fused_recurrent_gated_delta_rule.py [hunk 3]
```diff
@@ -1,10 +1,11 @@
...
 
 
 def _maybe_l2norm(x: torch.Tensor, enabled: bool) -> torch.Tensor:
     if not enabled:
         return x
-    return x / (torch.sqrt(torch.sum(x * x, dim=-1, keepdim=True)) + 1e-6)
+    return F.normalize(x, p=2, dim=-1).to(x.dtype)
```
[the remark this answers is not in the document]
Comment (Tflowers-0129) [reply vllm_ascend/_310p/ops/fla/fused_recurrent_gated_delta_rule.py]: updated, i have used the : F.normalize(key, p=2, dim=-1, eps=1e-6).to(key.dtype)
